include/zephyr/posix/unistd.h

```diff
@@ -11,16 +11,208 @@
 #ifdef CONFIG_NETWORKING
 /* For zsock_gethostname() */
 #include <zephyr/net/socket.h>
+#include <zephyr/net/hostname.h>
 #endif
 
 #ifdef CONFIG_POSIX_API
 #include <zephyr/fs/fs.h>
 #endif
 
+#ifdef CONFIG_POSIX_SYSCONF
+#include <zephyr/posix/signal.h>
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/* Version test macros */
+#define _POSIX_VERSION 200809L
+#define _POSIX2_VERSION (-1L)
+#define _XOPEN_VERSION (-1L)
+
+/* Internal helper macro to set constant if required Kconfig symbol is enabled */
+#define Z_SC_VAL_IFDEF(_def, _val) COND_CODE_1(_def, (_val), (-1L))
+
+/* Constants for Opitions and Option Groups */
+#define _POSIX_ADVISORY_INFO		  (-1L)
+#define _POSIX_ASYNCHRONOUS_IO		  _POSIX_VERSION
+#define _POSIX_BARRIERS			  Z_SC_VAL_IFDEF(CONFIG_PTHREAD_IPC, _POSIX_VERSION)
+#define _POSIX_CHOWN_RESTRICTED		  (-1L)
+#define _POSIX_CLOCK_SELECTION		  Z_SC_VAL_IFDEF(CONFIG_POSIX_CLOCK, _POSIX_VERSION)
+#define _POSIX_CPUTIME			  (-1L)
+#define _POSIX_FSYNC			  (-1L)
+#define _POSIX_IPV6			  Z_SC_VAL_IFDEF(CONFIG_NET_IPV6, _POSIX_VERSION)
+#define _POSIX_JOB_CONTROL		  (-1L)
+#define _POSIX_MAPPED_FILES		  _POSIX_VERSION
```

Comment:
Not sure about that but the standard says that it
 > shall always be set to the value 200809L.

Reply:
We don't support it yet unfortunately, so it should be -1L